books/kestrel/crypto/r1cs/sparse/p1cs.lisp

```diff
@@ -196,7 +240,23 @@
         ; named "1"
       )))
 
+(defthm p1cs-coefficient-returns-string
+  (implies (and (member fp (list (fp1) (fp2)))
+                (well-formed-term-p term))
+           (stringp (p1cs-coefficient fp term)))
+  :hints (("Goal" :in-theory (disable ; p1cs-coefficient
+                                      p1cs-negative
+                                      well-formed-term-p
+                                      assoc-equal))))
+
 (defun p1cs-term (fp term)
+  (declare (xargs :guard (and (member fp (list (fp1) (fp2)))
+                              (well-formed-term-p term))
+                  :guard-hints (("Goal" :in-theory (disable p1cs-coefficient
+                                                            p1cs-negative
+                                                            assoc-equal
+                                                            well-formed-term-p)))
+                  :guard-debug t))
```

Comment:
It is best to remove `:guard-debug t` after you've succeeded in proving the guard.